lib/is-valid-turn.js

```diff
@@ -0,0 +1,148 @@
+"use strict";
+
+const isValidMove = require("./is-valid-move");
+const makeMove = require("./make-move");
+const _ = require("lodash");
+const getAceyDeucey = require("./get-acey-deucey");
+const checkForWinner = require("./check-for-winner");
+const findAvailableSpaces = require("./find-available-spaces");
+
+function isValidTurn(gameState, diceRoll, proposedMoves) {
+    function proposedMovesMatchDice() {
+        if (diceRoll.length > 2) {
+                      
+            const aceyDeucey = getAceyDeucey(diceRoll);
+            if (aceyDeucey.isAceyDeucey === false) {
+                return false;
+            }
+            
+            const sortedAceyDeuceyRoll = _.sortBy(diceRoll);
+            _(3).range().forEach(() => sortedAceyDeuceyRoll.push(aceyDeucey.doublesVal));
+
+            if (!_(proposedMoves).take(2).map("numberOfSpaces").sortBy().isEqual([1, 2])) {
+                if (proposedMoves.length === 0) {
+                    const aceySpaces = findAvailableSpaces(gameState, 1);
+                    const deuceySpaces = findAvailableSpaces(gameState, 2);
+                
+                    return !aceySpaces.length && !deuceySpaces.length;
+                }
+                
+                if ([1, 2].indexOf(proposedMoves[0].numberOfSpaces) < 0) {
+                    return false;
+                }
+                
+                if (!isValidMove(gameState, proposedMoves[0])) {
+                    return false;
+                }
+                
+                const newGameState = makeMove(gameState, proposedMoves[0]);
+                
+                const nextRoll = proposedMoves[0].numberOfSpaces === 1 ? 2 : 1;
+                const nextAvailableSpaces = findAvailableSpaces(newGameState, nextRoll);
+                return !nextAvailableSpaces.length;
+            }
+            
+            if (proposedMoves.length < sortedAceyDeuceyRoll.length) {
+                let newGameState = gameState;
+                proposedMoves.forEach(move => {
+                    if (isValidMove(newGameState, move)) {
+                        newGameState = makeMove(newGameState, move);
+                    }
+                    return false;
```

Comment:
Bump; you successfully changed the `forEach` to a `for of` in another place, but we've got the same issue here. Maybe  it would be good to extract this shared functionality into a function?